### PR TITLE
INT-3163 - Add --api-base-url for sync, run, and validate-question CLI commands

### PR DESCRIPTION
**Pull request by adam-in-ict**
Add --api-base-url for sync, run, and validate-question CLI commands.

### CODE REVIEW

File: packages/integration-sdk-cli/src/commands/run.ts [hunk 1]
```diff
@@ -55,7 +56,19 @@ export function run() {
       log.debug('Loading account from JUPITERONE_ACCOUNT environment variable');
       const account = getAccountFromEnvironment();
 
-      const apiBaseUrl = getApiBaseUrl({ dev: options.development });
+      let apiBaseUrl: string;
```
Comment (adam-in-ict) [inline packages/integration-sdk-cli/src/commands/run.ts:59]: This overall logic is used in all three places with the only difference being what `development` mechanism is checked and what error is thrown.  We might look to try and standardize into a helper function when we come back through to add this functionality to additional commands.

File: docs/integrations/development.md [hunk 2]
```diff
@@ -1305,6 +1305,14 @@ the job run. The polling will stop once the job was marked as complete.
 
 ex: `j1-integration sync --instance <integration instance id> --tail`
 
+###### Option `--api-base-url`
+
+If provided this option specifies which base URL to use for synchronization.
+Cannot be used with the --development flag.
```
Comment (austinkelleher) [inline docs/integrations/development.md:1311]: Nit:

```suggestion
Cannot be used with the `--development` flag.
```

File: packages/integration-sdk-cli/src/commands/run.ts [hunk 3]
```diff
@@ -55,7 +56,19 @@ export function run() {
...
+      if (options.apiBaseUrl) {
+        if (options.development) {
```
Comment (austinkelleher) [inline packages/integration-sdk-cli/src/commands/run.ts:61]: Super minor nit, but we could reduce the nesting by slightly refactoring to something like:

```ts
if (options.apiBaseUrl && options.development) {
  throw new Error(
    'Invalid configuration supplied.  Cannot specify both --api-base-url and --development(-d) flags.',
  );
}

const apiBaseUrl = options.apiBaseUrl || getApiBaseUrl({
  dev: options.development,
});
```

We could also create a function:

(NOTE: would be nice if `options` was typed, but we could address that later)

```ts
function getApiBaseUrlFromCliArgs(options: any): string {
  if (options.apiBaseUrl && options.development) {
    throw new Error(
      'Invalid configuration supplied.  Cannot specify both --api-base-url and --development(-d) flags.',
    );
  }

  return options.apiBaseUrl || getApiBaseUrl({
    dev: options.development,
  });
}
```

**Review by austinkelleher — APPROVED** [review]
Approved with minor suggestions

**Pull request merged**